theories/ring.elpi

```diff
@@ -163,14 +164,16 @@ quote.zmod U R Morph {{ @intmul lp:U' lp:In1 lp:In2 }}
   quote.zmod U R Morph In1 OutM1 Out1 VarMap, !,
   quote.ring
     {{ int_Ring }} none R
-    (n\ {{ @intmul (GRing.Ring.zmodType lp:R) (@GRing.one lp:R) lp:n }})
+    (n\ {{ @intmul (GRing_Ring__to__GRing_Zmodule lp:R)
+                   (@GRing.one lp:R) lp:n }})
```

Comment:
`R` is a `ringType` instance but `GRing.one` takes a `semiRingType` instance now. This is a good example showing that we should eventually semi-automatically generate the reification procedure.

Reply:
Indeed, or at least factor it enough that we avoid duplicating those bugs.